types/json_printing.h

```diff
@@ -215,6 +215,25 @@ struct StringJsonPrintingContext
     void write(const std::string & s);
 };
 
+/*****************************************************************************/
+/* UTF8 STRING JSON PRINTING CONTEXT                                         */
+/*****************************************************************************/
+
+/** Writes a JSON representation to the given Utf8 string.  Note that the
+    string CANNOT be modified during writing; direct access to it is
+    undefined.
+
+    TODO: we should change this interface (and that of the previous class)
+    to have a str() method rather than pass in its result by reference.
+*/
+
+struct Utf8StringJsonPrintingContext
+    : public StringJsonPrintingContext {
+
+    Utf8StringJsonPrintingContext(Utf8String & str);
+    Utf8String & str;
```

Comment:
why not have this const & and the constructor taking a const & ? instead of the cast_const?